File: my-submissions/e1502.c
// https://leetcode.com/problems/can-make-arithmetic-progression-from-sequence/description/

int compareHelper(const void* one, const void* two) {
    return *((int *) one) - *((int *) two);
}

bool canMakeArithmeticProgression(int* arr, int arrSize) {
    qsort(arr, arrSize, sizeof(int), compareHelper);

    for (int i = 0; i < arrSize - 2; i++) {
        if (!(arr[i] - arr[i + 1] == arr[i + 1] - arr[i + 2])) {
            return false;
        }
    }
    return true;
}

File: my-submissions/e1502.py
# https://leetcode.com/problems/can-make-arithmetic-progression-from-sequence/

class Solution:
    def canMakeArithmeticProgression(self, arr: List[int]) -> bool:
        arr.sort()
        for i in range(len(arr) - 2) :
            if not arr[i] - arr[i + 1] == arr[i + 1] - arr[i + 2] :
                return False

        return True